ooni/geoip.py

```diff
@@ -198,23 +199,36 @@ def resolveGeodata(self,
             self.geodata['ip'] = '127.0.0.1'
 
     @defer.inlineCallbacks
+    def dns_lookup(self, dns_discovery='whoami.akamai.net'):
+        try:
+            answers = yield dns_client.lookupAddress(dns_discovery)
+            address = answers[0][0].payload.dottedQuad()
+        except Exception as exc:
+            log.exception(exc)
+            log.err("Failed to lookup the resolver IP address")
+            address = None
+        defer.returnValue(address)
+
+    @defer.inlineCallbacks
     def lookup(self, include_ip=None, include_asn=None, include_country=None):
         if self._state == IN_PROGRESS:
             yield self._looking_up
         elif self._last_lookup < time.time() - self._expire_in:
             self.address = None
 
         if self.address:
+            dns_address = yield self.dns_lookup()
             self.resolveGeodata(include_ip, include_asn, include_country)
-            defer.returnValue(self.address)
+            defer.returnValue((self.address, dns_address))
         else:
             self._state = IN_PROGRESS
             try:
                 yield self.askTor()
                 log.msg("Found your IP via Tor")
                 self.resolveGeodata(include_ip, include_asn, include_country)
                 self._looking_up.callback(self.address)
-                defer.returnValue(self.address)
+                dns_address = yield self.dns_lookup()
+                defer.returnValue((self.address, dns_address))
```

Comment:
To avoid duplicating this in multiple occasions, you probably want to re-arrange a bit the logic so that you only call `self.dns_lookup()` in one place and change the logic by which `defer.returnValue()` is called.